app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -120,26 +123,33 @@ public static void showInsertTableRowDialog(final Activity activity, final boole
         SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
     }
 
-    public static void showSearchFilesDialog(Activity activity, File searchDir, Callback.a1<String> callback) {
-        SearchOrCustomTextDialog.DialogOptions dopt = new SearchOrCustomTextDialog.DialogOptions();
-        baseConf(activity, dopt);
-        dopt.callback = query -> SearchOrCustomTextDialog.recursiveFileSearch(activity, searchDir, query, (Callback.a1<List<String>>) searchResults -> {
-            dopt.callback = callback;
-            dopt.isSearchEnabled = !searchResults.isEmpty();
-            dopt.data = searchResults;
-            dopt.cancelButtonText = R.string.close;
-            dopt.titleText = R.string.select;
-            dopt.messageText = null;
-            if (dopt.data.isEmpty()) {
-                dopt.messageText = "     ¯\\_(ツ)_/¯     ";
-            }
-            SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
-        });
-        dopt.titleText = R.string.search;
-        dopt.isSearchEnabled = true;
-        dopt.messageText = activity.getString(R.string.recursive_search_in_current_directory);
-        dopt.searchHintText = R.string.search;
-        SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
+    public static void showSearchFilesDialog(Activity activity, File searchDir, Callback.a2<String, Integer> callback) {
+        if (SearchEngine.isSearchExecuting) {
+            return;
+        }
+
+        AppSettings appSettings = new AppSettings(activity);
+
+        Callback.a1<FileSearchDialog.CallBackOptions> fileSearchDialogCallback = (cb_options) -> {
+
+            final boolean isShowResultOnCancel = appSettings.isShowSearchResultOnCancel();
+            final Integer maxSearchDepth = appSettings.getSearchMaxDepth();
+            final List<String> ignoredDirs = appSettings.getIgnoredSearchDirNames();
+            final List<String> ignoredFiles = appSettings.getIgnoredSearchFileNames();
+            final boolean isShowMatchPreview = appSettings.isShowMatchPreview();
+            final List<String> searchByContentExtensions = appSettings.getSearchByContentExtensions();
+            appSettings.setSearchQueryRegexUsing(cb_options._isRegexQuery);
```

Comment:
Do we really want to remember these options for next time? @gsantner

Reply:
I think we should carefully think what all settings we want to have. however putting PR review focus atm on the funcationalties and not breaking existing code

Reply:
:+1